airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,165 @@
+            ],
+    )
+    lateinit var configFile: Path
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_CATALOG_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_CATALOG_DESC,
+                "Required by the following commands: read, write",
+            ],
+    )
+    lateinit var catalogFile: Path
+
+    @CommandLine.Option(
+        names = ["--" + JavaBaseConstants.ARGS_STATE_KEY],
+        description =
+            [
+                JavaBaseConstants.ARGS_PATH_DESC,
+                "Required by the following commands: read",
+            ],
+    )
+    lateinit var stateFile: Optional<Path>
+
+    @CommandLine.Spec lateinit var commandSpec: CommandLine.Model.CommandSpec
+
+    override fun run() {
+        val result = execute(command)
+        result.onSuccess { airbyteMessage ->
+            airbyteMessage?.let { outputRecordCollector.accept(airbyteMessage) }
+        }
+        result.onFailure {
+            commandSpec.commandLine().usage(System.out)
+            logger.error(it) { "\nUnable to perform operation '$command'." }
+            // Many of the exceptions thrown are nested inside layers of RuntimeExceptions. An
+            // attempt is made
+            // to
+            // find the root exception that corresponds to a configuration error. If that does not
+            // exist, we
+            // just return the original exception.
+            ApmTraceUtils.addExceptionToTrace(it)
+            val rootThrowable = ConnectorExceptionUtil.getRootConfigError(Exception(it))
+            val displayMessage = ConnectorExceptionUtil.getDisplayMessage(rootThrowable)
+            // If the source connector throws a config error, a trace message with the relevant
+            // message should
+            // be surfaced.
+            if (ConnectorExceptionUtil.isConfigError(rootThrowable)) {
+                AirbyteTraceMessageUtility.emitConfigErrorTrace(it, displayMessage)
+            }
+            if (OperationType.CHECK.name.equals(command, true)) {
+                // Currently, special handling is required for the CHECK case since the user display
+                // information in
+                // the trace message is
+                // not properly surfaced to the FE. In the future, we can remove this and just throw
+                // an exception.
+                outputRecordCollector.accept(
+                    AirbyteMessage()
+                        .withType(AirbyteMessage.Type.CONNECTION_STATUS)
+                        .withConnectionStatus(
+                            AirbyteConnectionStatus()
+                                .withStatus(AirbyteConnectionStatus.Status.FAILED)
+                                .withMessage(displayMessage),
+                        ),
+                )
+            }
+        }
+
+        logger.info { "Completed integration: $connectorName" }
```

Comment:
What purpose does this log message serve? Is this expected?

Reply:
@postamar This is copied from the `IntegrationRunner` logic. I have no idea if it serves any purpose other than to mark the end of the execution.